### ISSUE TITLE
NullPointerException in WaitForServiceDiscoveryReq

### ISSUE THREAD

**Issue by lategoodbye** [V2GClarity/RISE-V2G#4]
Using latest RiseV2G as EVSE and a different ISO 15118 implementation as EV i'm able to trigger a NullPointerException in WaitForServiceDiscoveryReq:

```
2017-07-27T17:02:52,893 DEBUG [ConnectionThread fe80:0:0:0:595e:5080:49ac:ab38%2] V2GCommunicationSessionSECC: New state is WaitForServiceDiscoveryReq
2017-07-27T17:02:52,893 DEBUG [ConnectionThread fe80:0:0:0:595e:5080:49ac:ab38%2] ConnectionHandler: Message received
2017-07-27T17:02:52,910 DEBUG [ConnectionThread fe80:0:0:0:595e:5080:49ac:ab38%2] OpenEXICodec: XML representation of ServiceDiscoveryReq:
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><ns6:V2G_Message xmlns:ns6="urn:iso:15118:2:2013:MsgDef" xmlns:ns5="http://www.w3.org/2000/09/xmldsig#" xmlns:ns7="urn:iso:15118:2:2013:MsgBody" xmlns:ns2="urn:iso:15118:2:2010:AppProtocol" xmlns:ns4="urn:iso:15118:2:2013:MsgDataTypes" xmlns:ns3="urn:iso:15118:2:2013:MsgHeader"><ns6:Header><ns3:SessionID>E75C624A5F7A76C4</ns3:SessionID></ns6:Header><ns6:Body><ns7:ServiceDiscoveryReq><ns7:ServiceCategory>EVCharging</ns7:ServiceCategory></ns7:ServiceDiscoveryReq></ns6:Body></ns6:V2G_Message>
2017-07-27T17:02:52,911 DEBUG [ConnectionThread fe80:0:0:0:595e:5080:49ac:ab38%2] WaitForServiceDiscoveryReq: ServiceDiscoveryReq received
Exception in thread "ConnectionThread fe80:0:0:0:595e:5080:49ac:ab38%2" java.lang.NullPointerException
	at org.eclipse.risev2g.secc.states.WaitForServiceDiscoveryReq.getServiceList(WaitForServiceDiscoveryReq.java:119)
	at org.eclipse.risev2g.secc.states.WaitForServiceDiscoveryReq.processIncomingMessage(WaitForServiceDiscoveryReq.java:52)
	at org.eclipse.risev2g.secc.session.V2GCommunicationSessionSECC.processIncomingMessage(V2GCommunicationSessionSECC.java:183)
	at org.eclipse.risev2g.secc.session.V2GCommunicationSessionSECC.update(V2GCommunicationSessionSECC.java:161)
	at java.util.Observable.notifyObservers(Observable.java:159)
	at org.eclipse.risev2g.secc.transportLayer.ConnectionHandler.run(ConnectionHandler.java:134)
	at java.lang.Thread.run(Thread.java:748)
```

According to the code at that place it looks like a mix-up of serviceCategoryFilter and serviceScopeFilter:

```
if (serviceCategoryFilter != null)
			getLogger().debug("EVCC filters offered services by category: " + serviceScopeFilter.toString());
```

Since i'm not absolutely sure about the right fix, i created this issue rather than a pull request.

**Comment by MarcMueltin** [comment]
You are absolutely right, there is a mix up of of serviceCategoryFilter and serviceScopeFilter. So it should be:

```
if (serviceCategoryFilter != null)
			getLogger().debug("EVCC filters offered services by category: " + serviceCategoryFilter.toString()); 
```

I corrected it and will publish it with the next release.

**Comment by MarcMueltin** [comment]
Issue resolved with commit f301e82.
